packages/modules/analytics/src/services/analytics-service.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  TrackAnalyticsEventDTO,
+  IdentifyAnalyticsEventDTO,
+} from "@medusajs/types"
+import AnalyticsProviderService from "./provider-service"
+
+type InjectedDependencies = {
+  analyticsProviderService: AnalyticsProviderService
+}
+
+export default class AnalyticsService {
+  protected readonly analyticsProviderService_: AnalyticsProviderService
+
+  constructor({ analyticsProviderService }: InjectedDependencies) {
+    this.analyticsProviderService_ = analyticsProviderService
+  }
+
+  __hooks = {
+    onApplicationShutdown: async () => {
+      this.analyticsProviderService_.shutdown()
+    },
+  }
+
+  getProvider() {
+    return this.analyticsProviderService_
+  }
+
+  async track(data: TrackAnalyticsEventDTO): Promise<void> {
+    this.analyticsProviderService_.track(data)
```

Comment:
suggestion: since all the methods are async but we don't await or return the promise to be resolve, should we handle the errors in case there is any? Just to not have unhandled promises

Reply:
Adding await, and I'll also add a try-catch to make the error nicer

Reply:
that sounds good to me 👍